inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -124,8 +124,8 @@ def arxiv_plot_extract(obj, eng):
     tarball = obj.files[filename]
 
     if tarball:
-        with TemporaryDirectory(prefix='plot_extract') as scratch_space:
-            tarball_file = retrieve_uri(tarball.file.uri, outdir=scratch_space)
+        with TemporaryDirectory(prefix='plot_extract') as scratch_space, \
+                retrieve_uri(tarball.file.uri, outdir=scratch_space) as tarball_file:
```

Comment:
Interesting, I did not know that you could use the `as` from the first element in the second element.

Reply:
Indeed, from https://docs.python.org/2/reference/compound_stmts.html#the-with-statement:
>With more than one item, the context managers are processed as if multiple with statements were nested:
>
>```
>with A() as a, B() as b:
>    suite
>```
>
>is equivalent to
>
>```
>with A() as a:
>    with B() as b:
>        suite
>```